exe/cli/cmd/deals_watch.go

```diff
@@ -0,0 +1,125 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/gosuri/uilive"
+	"github.com/ipfs/go-cid"
+	"github.com/spf13/cobra"
+	"github.com/textileio/filecoin/api/client"
+)
+
+// var DealStates = []string{
+// 	"DealUnknown",
+// 	"DealRejected",
+// 	"DealAccepted",
+// 	"DealStaged",
+// 	"DealSealing",
+// 	"DealFailed",
+// 	"DealComplete",
+// 	"DealError",
+// }
+
+func init() {
+	watchCmd.Flags().StringSliceP("cids", "c", []string{}, "List of deal cids to watch")
+
+	dealsCmd.AddCommand(watchCmd)
+}
+
+var watchCmd = &cobra.Command{
+	Use:   "watch",
+	Short: "Watch storage process",
+	Long:  `Watch storage process`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithTimeout(context.Background(), cmdTimeout)
+		defer cancel()
+
+		cidStrings, err := cmd.Flags().GetStringSlice("cids")
+		checkErr(err)
+		if len(cidStrings) == 0 {
+			Fatal(errors.New("you must provide at least one cid using the cids flag"))
+		}
+
+		cids := make([]cid.Cid, len(cidStrings))
+		for i, s := range cidStrings {
+			cid, err := cid.Parse(s)
+			checkErr(err)
+			cids[i] = cid
+		}
+
+		state := make(map[string]*client.WatchEvent, len(cids))
+		for _, cid := range cids {
+			state[cid.String()] = nil
+		}
+
+		writer := uilive.New()
+		writer.Start()
+
+		updateOutput(writer, state)
+
+		ch, err := fcClient.Deals.Watch(ctx, cids)
+		checkErr(err)
+
+		for {
+			event, ok := <-ch
+			if ok == false {
+				break
+			}
+			state[event.Deal.ProposalCid.String()] = &event
+			updateOutput(writer, state)
+			if isComplete(state) {
+				break
+			}
+		}
```

Comment:
Maybe this code could be extracted and be used by other commands that call `Watch`, like in the interactive cmd.
If that doesn't sound like may worth it; same comments as in the other case (`!ok` and error msg)

Reply:
Extracted it.